### PR TITLE
Ignore spurious diffs in cpu resource requests/limits

### LINKED ISSUE #33: Spurious kubediff due to CPU resources
We are currently seeing this kubediff on prod:

```
## scope/pipe (Deployment)

.spec.template.spec.containers: 'element [0]' missing
## monitoring/alertmanager (Deployment)

.spec.template.spec.containers: 'element [0]' missing
```

AFAICT, this is because for scope/pipe, the YAML file says:

```yaml
        resources:
          requests:
            cpu: 0.5
```

But the k8s object says:

```yaml
        resources:
          requests:
            cpu: 500m
```

Despite being freshly applied.

Since this difference isn't meaningful, it shouldn't be reported.

### PR DESCRIPTION
**Pull request by ekimekim**
There are multiple possible formats for the same semantic value (eg. 50m vs 0.05)
for cpu resources, and the k8s server may return one even if you apply the other.
This creates a diff that can't be resolved by applying.

We solve this problem by ignoring the diff if the values have the same meaning.
To facilitate this we introduce a generic method for adding 'tolerations' to the diffs.
Since it would be complex and annoying to add the special case all the way down the stack
at the place the diff is detected, we instead remove any tolerable diffs as a seperate step
after all diffs for a file have been generated.

I've used globbing of paths for simplicity, as it allows you to easily say "anything ending with"
or "all paths under", etc. without resorting to full regex, which I think would be less readable
in most cases.

Fixes #33

### CODE REVIEW

**Comment by ekimekim** [comment]
on further testing, it looks like this approach doesn't work - `diff_lists` relies on `diff()` directly to check equality and treats items as 'missing' rather than propagating the underlying diffs. So we need to check tolerations at `diff()` time.

**Comment by ekimekim** [comment]
Ok, refactored to do it slightly differently. Working now.

File: kubedifflib/_diff.py [hunk 1]
```diff
@@ -79,7 +97,7 @@ def diff_dicts(path, want, have):
 
 
 def normalize(value):
-  if isinstance(value, int):
+  if isinstance(value, (int, float)):
```
Comment (jml) [inline kubedifflib/_diff.py]: I think we can do `isinstance(value, numbers.Number)` (as per https://docs.python.org/2/library/numbers.html)

**Comment by jml** [comment]
@ekimekim ping?

**Pull request merged**